### PR TITLE
refactor: apply `webpack-defaults`

### LINKED ISSUE #24: webpack-defaults upgrade
Addition of `webpack-defaults` & associated refactoring as a part of the next `Major` release

Issue exists for status tracking across the organization. 

>Please do not close

### PR DESCRIPTION
**Pull request by bebraw**
Closes #24.

### CODE REVIEW

**Comment by bebraw** [comment]
I updated it to webpack-defaults 1.1. I don't mind bumping later again.

The only thing I noticed is that it overrode ESLint completely. I had to disable a couple of rules to accommodate to the project so those were lost. Is this by design?

**Comment by Kovensky** [comment]
Yes, webpack-defaults always overrides the eslint settings; presumably to install new rules if any are added to defaults.

Not sure if it's intentional that it doesn't support customizations at all.

**Comment by bebraw** [comment]
@Kovensky Ok, track https://github.com/webpack-contrib/webpack-defaults/issues/59 .

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      webpack_version: 2.6.0
+      job_part: test
+    - nodejs_version: '4.3'
+      webpack_version: 2.6.0
+      job_part: test
+build: 'off'
+matrix:
+  fast_finish: true
+install:
+  - 'ps: Install-Product node $env:nodejs_version x64'
```
Comment (d3viant0ne) [inline appveyor.yml]: You have to strip the `'` off of anything that has `ps: || cmd:`. Already has an open issue in defaults.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,34 @@
+sudo: false
```
Comment (d3viant0ne) [inline .travis.yml]: Pickup `webpack-defaults@1.4.0` for all the psudo canary Travis changes.

File: package.json [hunk 3]
```diff
@@ -56,23 +69,40 @@
...
+    "del-cli": "^1.0.0",
     "eslint": "^3.13.1",
     "eslint-config-airbnb": "^14.0.0",
+    "eslint-config-webpack": "^1.2.3",
     "eslint-plugin-import": "^2.2.0",
     "eslint-plugin-jsx-a11y": "^3.0.2",
     "eslint-plugin-node": "^4.0.1",
     "eslint-plugin-react": "^6.9.0",
     "git-prepush-hook": "^1.0.1",
     "jest": "^18.1.0",
```
Comment (d3viant0ne) [inline package.json]: Change to `"jest": "^20.0.4",`,

File: package.json [hunk 4]
```diff
@@ -56,23 +69,40 @@
     "babel-jest": "^18.0.0",
```
Comment (d3viant0ne) [inline package.json]: Change to `"babel-jest": "^20.0.3",`

File: package.json [hunk 5]
```diff
@@ -56,23 +69,40 @@
...
     "babel-plugin-syntax-object-rest-spread": "^6.13.0",
     "babel-plugin-transform-object-rest-spread": "^6.20.2",
+    "babel-polyfill": "^6.23.0",
+    "babel-preset-env": "^1.5.1",
     "babel-preset-es2015": "^6.18.0",
+    "cross-env": "^5.0.0",
+    "del-cli": "^1.0.0",
     "eslint": "^3.13.1",
```
Comment (d3viant0ne) [inline package.json]: Strip the eslint deps in favor of ..

```
    "eslint": "^3.19.0",
    "eslint-config-webpack": "^1.2.3",
    "eslint-plugin-import": "^2.3.0",
```

File: package.json [hunk 6]
```diff
@@ -56,23 +69,40 @@
...
     "eslint-config-airbnb": "^14.0.0",
+    "eslint-config-webpack": "^1.2.3",
     "eslint-plugin-import": "^2.2.0",
     "eslint-plugin-jsx-a11y": "^3.0.2",
     "eslint-plugin-node": "^4.0.1",
     "eslint-plugin-react": "^6.9.0",
     "git-prepush-hook": "^1.0.1",
```
Comment (d3viant0ne) [inline package.json]: Shouldn't need this anymore?

File: package.json [hunk 7]
```diff
@@ -56,23 +69,40 @@
     "babel-jest": "^18.0.0",
     "babel-plugin-syntax-object-rest-spread": "^6.13.0",
     "babel-plugin-transform-object-rest-spread": "^6.20.2",
+    "babel-polyfill": "^6.23.0",
```
Comment (d3viant0ne) [inline package.json]: Strip all the Babel in favor of ...

```
    "babel-cli": "^6.24.1",
    "babel-jest": "^20.0.3",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-polyfill": "^6.23.0",
    "babel-preset-env": "^1.5.2",
```

File: package.json [hunk 8]
```diff
@@ -56,23 +69,40 @@
...
     "git-prepush-hook": "^1.0.1",
     "jest": "^18.1.0",
+    "lint-staged": "^3.5.0",
+    "nsp": "^2.6.3",
+    "pre-commit": "^1.2.2",
+    "standard-version": "^4.0.0",
     "sync-exec": "^0.6.2",
     "uglify-js": "^2.8.18",
-    "webpack": "^2.2.0"
+    "webpack": "^2.2.0",
```
Comment (d3viant0ne) [inline package.json]: Pull the hard devDependency on `webpack`

**Comment by d3viant0ne** [comment]
@Kovensky - `webpack-defaults always overrides the eslint settings` is very intentional. 

The entire point of defaults was to standardize all of the loaders & plugins in contrib to simplify contributions & maintenance, this includes amongst other things, code style. We have a single way of doing things because in the end, it's always going to be a small group of people handling a vast majority of the project maintenance. 

Rules can be overridden in the `.eslintrc` if it's necessary or a pull request can be opened in `eslint-config-webpack` for rules that should be changed across the board.

There is discussion about using Prettier in the future but as it stands right now, there is far more value in getting the standards work done than tinkering around with stylistic debates. We can take another stylistic pass, probably with prettier in webpac-defaults 2.x along with the schema-utils updates but until https://github.com/orgs/webpack-contrib/projects/5 board is cleared, we are going with `good enough`

**Comment by michael-ciniawsky** [comment]
Updated to `webpack-defaults v1.4.0` can I merge it so we can start backporting the test and update to uglify >= v3.0.0` ?

**Comment by michael-ciniawsky** [comment]
Why is the CI not working? 😕 😭

**Comment by TheLarkInn** [comment]
Shipit.

**Pull request merged**